app/models/piece.rb

```diff
@@ -36,7 +44,11 @@ def captured?
   def valid_move?(to_row, to_col)
     return false if move_nil?(to_row, to_col)
     return false if move_out_of_bounds?(to_row, to_col)
-    return false if move_destination_same_side?(to_row, to_col)
+    return false if move_destination_ally?(to_row, to_col)
+
+    # until have Knight::move_legal?
+    return true if type == "Knight"
```

Comment:
I just wrote a quick Knight:: move_legal? which work. I will push it up in a few.

The rubocop complexity issues still remains. Perhaps someone else can get rid of it. We do not want to check for move_obstructed? for a Knight. They can jump so their path is not obstructed. If I move the type != "Knight" into move_obstructed? I get the complexity error there.